### PR TITLE
More on tutorial

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Removed a largish block of repeated text.
- Added sections "Top List and Base List" and "Methods for Defining Options" (on, define, etc.).
- Linked from class OptionParser doc to the tutorial.

I'm not positive that we need the two sections, but it took be some work to learn and write, so let's consider them.

### CODE REVIEW

File: doc/optparse/ruby/basic.rb [hunk 1]
```diff
@@ -12,5 +12,6 @@
 parser.on('-z', 'Whether to Z') do |value|
   p ['z', value]
 end
-# Parse the command line.
-parser.parse!
+# Parse the command line and return pared-down ARGV.
```
Comment (olleolleolle) [inline doc/optparse/ruby/basic.rb:15]: Can we find a word which is more common than pared-down? Here, I'm thinking of readers whose not first-language English.

I see the specificity, though. Terse etc. 

(Oh, I'll get back to this, gotta sleep!)

**Comment by BurdetteLamar** [comment]
@olleolleolle, will you be looking at this?

**Comment by BurdetteLamar** [comment]
@nobu, can you take a look at this PR?  (I have a biggish block of RDoc about parsing that's waiting for this PR to clear.)   Thanks.

**Comment by olleolleolle** [comment]
@BurdetteLamar It's been a bit hectic, here and I've been far from computer screens when not at work. Soon!

File: doc/optparse/tutorial.rdoc [hunk 2]
```diff
@@ -597,3 +604,57 @@ Execution:
...
+When \OptionParser builds its help text, the options in the top list
+precede those in the base list.
+
+=== Methods for Defining Options
+
+Option-defining methods allow you to create an option, and also append/prepend it
+to the top list or append it to the base list.
+
+Each of these next three methods accepts a sequence of parameter arguments and a block
+creates an option object using method \Option#make_switch (see below),
```
Comment (olleolleolle) [inline doc/optparse/tutorial.rdoc:628]: Does this sentence need a comma?

**Review by olleolleolle — APPROVED** [review]
I like the additional context added by this PR! 

The parts which say "these work like these other methods, but return differently" is especially good for the careful reader.

File: doc/optparse/tutorial.rdoc (hunk 2 above)
Comment (BurdetteLamar) [reply doc/optparse/tutorial.rdoc:628]: Yes, and fixed.

**Comment by BurdetteLamar** [comment]
Thanks, @olleolleolle, for your careful reviews.

**Pull request merged**